cfme/configure/access_control.py

```diff
@@ -332,3 +360,98 @@ def copy(self, name=None):
              action=form_buttons.add)
         flash.assert_success_message('Role "%s" was saved' % new_role.name)
         return new_role
+
+
+class Tenant(Updateable, Pretty):
+    """Kudos to mfalesni"""
+
+    save_changes = form_buttons.FormButton("Save changes")
+    tenant_form = Form(
+        fields=[
+            ('name', Input('name')),
+            ('description', Input('description'))
+        ])
+    pretty_attrs = ["name", "description"]
+
+    @classmethod
+    def get_root_tenant(cls):
+        return cls(name="My Company", _default=True)
+
+    def __init__(self, name=None, description=None, parent_tenant=None, _default=False):
+        self.name = name
+        self.description = description
+        self.parent_tenant = parent_tenant
+        self._default = _default
+
+    @property
+    def parent_tenant(self):
+        if self._default:
+            return None
+        if self._parent_tenant:
+            return self._parent_tenant
+        return self.get_root_tenant()
+
+    @parent_tenant.setter
+    def parent_tenant(self, tenant):
+        if tenant is not None and isinstance(tenant, Project):
+            # If we try to
+            raise ValueError("Project cannot be a parent object.")
+        if isinstance(tenant, basestring):
+            # If parent tenant is passed as string, we assume that tenant name was passed instead of object
+            tenant = Tenant(tenant)
+        self._parent_tenant = tenant
+
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return False
+        else:
+            return self.name == other.name
+
+    @property
+    def exists(self):
+        try:
+            sel.force_navigate("cfg_tenant_project", context={"tenant": self})
+        except CandidateNotFound:
+            return False
+        else:
+            return True
+
+    @property
+    def tree_path(self):
+        if self._default:
+            return [self.name]
+        else:
+            return self.parent_tenant.tree_path + [self.name]
+
+    @property
+    def parent_path(self):
+        return self.tree_path[:-1]
+
+    def create(self, cancel=False):
+        sel.force_navigate("cfg_tenant_project_create", context={"tenant": self})
+        fill(self.tenant_form, self, action=form_buttons.add)
+        if type(self) is Tenant:
+            flash.assert_success_message('Tenant "{}" was saved'.format(self.name))
+        elif type(self) is Project:
+            flash.assert_success_message('Project "{}" was saved'.format(self.name))
+        else:
+            raise TypeError('No Tenant or Project class passed to create method{}'.format(type(self).__name__))
+
+    def update(self, updates):
+        sel.force_navigate("cfg_tenant_project_edit", context={"tenant": self})
+        # Workaround 1
+        sel.wait_for_ajax()
+        # Workaround 2
```

Comment:
Add a comment that this particular workaround is due to the form appearing after a small delay.